### PR TITLE
Trait fixes and tweaks

### LINKED ISSUE #36085: Visual bug with new Trait system and Character Setup menu.
Player character has prosthetic limb(s) even without proper perk.  
bug

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: RDS now triggers twice as often and more intensely.
tweak: Night Vision is slightly brighter so that it's actually noticeable.
fix: Prosthetic limbs will no longer appear on the character setup even without having the trait.
/:cl:

Fixes #36085.

I'm very happy with the reception of traits so far. RDS and Night Vision aren't actually broken, but they *are* undertuned. Night Vision's impact is basically undetectable. As a result, I noticeably brightened Night Vision, and changed RDS to trigger with a 2% chance every second (instead of 1%) and to always give 50 hallucination if it didn't have a unique effect, instead of between 10 and 50.

### CODE REVIEW

File: code/datums/traits/_trait.dm [hunk 1]
```diff
@@ -48,6 +48,8 @@
 	if(QDELETED(trait_holder))
 		qdel(src)
 		return
+	if(istype(src, /mob/living/carbon/human/dummy))
```
Comment (SpaceManiac) [inline code/datums/traits/_trait.dm]: Probably want to check `trait_holder` here, not `src`.

Comment (Xhuis) [reply code/datums/traits/_trait.dm]: You are absolutely right!

**Comment by Dorsisdwarf** [comment]
Why is it not possible to edit your traits after a round has started, unlike every single other aspect of character creation?

**Comment by Xhuis** [comment]
Trait setups changing means that between clonings your traits could be modified, so you could say, remove your pacifism and start with something like RDS instead between bodies. I tried endlessly to find a workaround for this but encountered bizarre errors and failed workarounds, so I gave up. :v

**Comment by Dorsisdwarf** [comment]
I mean it seems to work fine for the rest of the game prefs...

**Comment by Xhuis** [comment]
No, this is was super odd. I was literally having it pass `if(FALSE)` checks and doing the things under them. I had no idea why it was happening. It's not the best code, because I had to work around my checks only sometimes doing what I wanted them to.

**Comment by Xhuis** [comment]
Am making a PR to change how traits are assigned and stored, will be closing this one when I pull it.

**Comment by Xhuis** [comment]
Closed in favor of #36160

**Pull request closed without merging**